File: common/src/main/java/dev/qixils/aviary/common/messaging/serializers/impl/StringSerializer.java
package dev.qixils.aviary.common.messaging.serializers.impl;

import dev.qixils.aviary.common.messaging.serializers.ByteSerializer;
import org.jetbrains.annotations.NotNull;

import java.nio.charset.StandardCharsets;

/**
 * A {@link ByteSerializer} for {@link String}s.
 */
public class StringSerializer implements ByteSerializer<String> {

	@Override
	public byte @NotNull [] serialize(@NotNull String string) {
		return string.getBytes(StandardCharsets.UTF_8);
	}

	@Override
	public @NotNull String deserialize(byte @NotNull [] data) {
		return new String(data, StandardCharsets.UTF_8);
	}
}

File: common/src/main/kotlin/dev/qixils/aviary/common/messaging/serializers/impl/StringSerializer.kt
package dev.qixils.aviary.common.messaging.serializers.impl

import dev.qixils.aviary.common.messaging.serializers.ByteSerializer
import java.nio.charset.StandardCharsets

/**
 * A [ByteSerializer] for [String]s.
 */
class StringSerializer : ByteSerializer<String> {

	override fun serialize(obj: String): ByteArray {
		return obj.toByteArray(StandardCharsets.UTF_8)
	}

	override fun deserialize(data: ByteArray): String {
		return String(data, StandardCharsets.UTF_8)
	}
}
